hypothesis-python/tests/cover/test_provisional_strategies.py

```diff
@@ -53,3 +56,16 @@ def test_is_IP6_addr(address):
     assert all(len(part) == 4 for part in as_hex)
     raw = unhexlify(address.replace(u":", u"").encode("ascii"))
     assert len(raw) == 16
+
+
+@pytest.mark.parametrize("max_length", [-1, 0, 3, 4.0, 256])
+@pytest.mark.parametrize("max_element_length", [-1, 0, 4.0, 64, 128])
+def test_invalid_domain_arguments(max_length, max_element_length):
+    with pytest.raises(InvalidArgument):
+        domains(max_length=max_length, max_element_length=max_element_length).example()
+
+
+@pytest.mark.parametrize("max_length", [None, 4, 8])
+@pytest.mark.parametrize("max_element_length", [None, 1, 2, 4, 8])
```

Comment:
Minor: let's get tests that we can go right up to the max value?

so a test for `max_length=255` and `max_element_length=63`.